devito/operator.py

```diff
@@ -253,10 +253,28 @@ def _prepare_arguments(self, **kwargs):
         Process runtime arguments passed to ``.apply()` and derive
         default values for any remaining arguments.
         """
-        # Process data-carriers (first overrides, then fill up with whatever is needed)
+        overrides, defaults = split(self.input, lambda p: p.name in kwargs)
+        # Process data-carrier overrides
         args = ReducerMap()
-        args.update([p._arg_values(**kwargs) for p in self.input if p.name in kwargs])
-        args.update([p._arg_values() for p in self.input if p.name not in args])
+        for p in overrides:
+            args.update(p._arg_values(**kwargs))
+            try:
+                args = ReducerMap(args.reduce_all())
+            except ValueError:
+                raise ValueError("Override `%s` is incompatible with overrides `%s`" %
+                                 (p, [i for i in overrides if i.name in args]))
```

Comment:
From my experience (may have changed) the % followed by line break fails to process the string expression

Reply:
But if test pass may have been fixed in python